broker-core/src/main/java/io/zeebe/broker/job/JobBatchActivateProcessor.java

```diff
@@ -161,8 +161,9 @@ private void activateJobs(TypedStreamWriter streamWriter, JobBatchRecord value)
       final JobRecord copiedJob = new JobRecord();
       copiedJob.wrap(copy, 0, jobRecord.getLength());
 
-      state.activate(key, copiedJob);
+      // first write follow up event as state.activate will clear the payload
       streamWriter.appendFollowUpEvent(key, JobIntent.ACTIVATED, copiedJob);
```

Comment:
On small question on this:

Do we really need the payload on `JobIntent.ACTIVATED`. I mean no one consumes this event. The job record with the collected payload is already in the JobBatch record.

Reply:
Good question, I'm also not sure. One of my earlier approaches actually removed the payload from the activated record. But I wasn't sure if this is a good idea from the exporter perspective. So I decided to keep it there to not break this behavior. Feel free to challenge that with others.

Reply:
More extremer question:

Do we have to write this event (Job ACTIVATED) at all, since it completely exist in the JobBatch record as well.

Reply:
@saig0 @npepinpe

Reply:
I think we kept it around initially because we were consuming it for something, but now I can't remember what it was. I can't think of a good argument to keep it around (other than it's a breaking change if anyone was actually consuming it). Maybe verify with Operate so they aren't caught by surprise (but I'm pretty sure they don't consume it).